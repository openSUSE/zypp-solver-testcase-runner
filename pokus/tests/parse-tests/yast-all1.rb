#!/usr/bin/ruby


# Check handling of various package data.


require 'zypptools/lib/package'
require 'zypptools/output/output'
include ZYppTools

require '../../src/testsuite'


packages = Array.new()

pkg = Package.new("test")
pkg.arch = "i586"
pkg.version = "1.2"
pkg.release = 1
pkg.installsize = 84000
pkg.rpmsize = 42000
pkg.summary = "A pure test package"
packages.push(pkg)


path = Testsuite::write_repo(:yast, packages)


Testsuite.set_arch("i586")
pool = Testsuite.read_repo("file://" + path)


Testsuite::haha2(pool).each do |res|
    puts "#{res.kind} #{res.name} #{res.edition.to_s} #{res.arch.to_s}"
    puts "  Summary: #{res.summary}"
    puts "  Size: #{res.size} / #{res.download_size}"
    Testsuite::dump_all_deps(res)
    puts
end

