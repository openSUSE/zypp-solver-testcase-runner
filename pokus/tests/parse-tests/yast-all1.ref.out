package test 1.2-1 i586
  Summary: A pure test package
  Size: 82.0 K / 41.0 K
  Provides: test == 1.2-1

product SUSE_LINUX 10.3 i586
  Summary: openSUSE 10.3
  Size: 0 B / 0 B
  Provides: SUSE_LINUX == 10.3
  Provides: openSUSE-10.3 == 10.3
  Requires: openSUSE-release-10.3

